src/Workspaces/Core/Portable/Workspace/Workspace_Events.cs

```diff
@@ -160,7 +160,7 @@ protected Task RaiseDocumentClosedEventAsync(Document document)
         /// An event that is fired when the active context document associated with a buffer 
         /// changes.
         /// </summary>
-        internal event EventHandler<DocumentEventArgs> DocumentActiveContextChanged
+        public event EventHandler<DocumentEventArgs> DocumentActiveContextChanged
```

Comment:
@CyrusNajmabadi, it is the same information that is sent along with the other public `Document` events (such as `DocumentOpened` and `DocumentClosed`). Any risk that would exist here also exists there, and modifying `DocumentEventArgs` to no longer expose the `Document` would be a breaking change at this point.

Looking at the code, the Document is for the new context (based on tracking the callers of `RaiseDocumentActiveContextChangedEventAsync` back to here http://source.roslyn.io/#Microsoft.VisualStudio.LanguageServices/Implementation/ProjectSystem/VisualStudioWorkspaceImpl.cs,884). Additionally, `DocumentClosed` is for the closed document and `DocumentOpen` is for the opened document.